Libraries/Image/RCTImageLoader.m

```diff
@@ -400,7 +400,13 @@ - (RCTImageLoaderCancellationBlock)loadImageOrDataWithURLRequest:(NSURLRequest *
             return;
           }
 
-          NSURL *redirectURL = [NSURL URLWithString: location];
+          NSURL *redirectURL;
+          if (![location hasPrefix:@"http"]) {
```

Comment:
@maartenschumacher you'll have to verify this, but I believe if you just write:

    [URLWithString:location relativeToURL:request.URL]

without checking the prefix, it will do the right thing anyway, i.e. if it's already an absolute URL it will ignore the `relativeToURL` argument.